kdc/krb5tgs.c

```diff
@@ -1671,6 +1675,54 @@ tgs_build_reply(astgs_request_t priv,
 		goto out;
 	    }
 
+	    /* Fetch the name from the TGT. */
+	    ret = _krb5_principalname2krb5_principal(context, &user2user_princ,
+						     adtkt.cname, adtkt.crealm);
+	    if (ret) {
+		goto out;
+	    }
+
+	    ret = krb5_unparse_name(context, user2user_princ, &user2user_name);
+	    if (ret) {
+		goto out;
+	    }
+
+	    /* Look up the name given in the TGT in the database. */
+	    ret = db_fetch_client(context, config, flags, user2user_princ, user2user_name,
+				  krb5_principal_get_realm(context, krbtgt_out->entry.principal),
+				  NULL, &user2user_client);
+	    if (ret) {
+		goto out;
+	    }
+
+	    if (user2user_client != NULL) {
+		/*
+		 * If the account is present in the database, check the account
+		 * flags.
```

Comment:
if `user2user_client == NULL`, what happens? I think we need to check `adtkt->cname` matches the request server.

Reply:
Other code rightly assumes that if `ret == 0` then the equivalent of `user2user_client` is non-NULL so I'll remove the check.

Reply:
Actually I stuffed up, we really don't want a case where `user2user_client == NULL` and so we have changed to not use the wrapper that allows `user2user_client == NULL`.  Thanks for the eagle eye!